qcelemental/models/align.py

```diff
@@ -126,6 +126,27 @@ def align_hessian(self, hess) -> NDArray:
         alhess = blockwise_contract(alhess)
         return alhess
 
+    def align_vector_gradient(self, mu_x, mu_y, mu_z):
+        """Align the nuclear gradients of vector components (e.g. dipole derivatives)."""
+        # Input data is assumed to be organized into separate x, y, z vector components.
+        # Organize derivatives for each atom into 3x3 and transform it.
+        nat = mu_x.shape[0]
+        al_mu_x = np.zeros( (nat, 3) )
+        al_mu_y = np.zeros( (nat, 3) )
+        al_mu_z = np.zeros( (nat, 3) )
+
+        Datom = np.zeros( (3,3) ) # atom whose nuclear derivatives are taken
+        for at in range(nat):
+            Datom.fill(0)
+            Datom[0,:] = mu_x[at,:]
+            Datom[1,:] = mu_y[at,:]
+            Datom[2,:] = mu_z[at,:]
+            Datom[:] = np.dot( self.rotation.T , np.dot(Datom, self.rotation) )
+            al_mu_x[self.atommap[at]][:] = Datom[0,:]
```

Comment:
does the Cfour atom ordering for `DIPDER` match the geometry and gradient in `GRD`? I earnestly hope so, as that's where `atommap` gets computed from.

Reply:
OK, yes they are the same.  Turns out that I was doing a bunch of second derivative computations and hadn't even noticed a GRD file being produced (in my scratch space).  Thus my concern.  A bit disturbing that the orientation can't be easily found in the output file, but it looks perhaps that CFOUR spits out the geometry to GRD even if it is not a gradient computation?  I will test the atom mapping now against the GRD coordinates and make sure it works.  Tricky for a user though who naively (naturally!) might think to provide the reoriented (but not reordered) geometry at the beginning of the output file.  Still, a step in the right direction.

Reply:
if all else fails, I think the atom mapping is in `JOBARC` (which we can read in binary). but I'd prefer to stick with the find-the-mapping-and-orientation tech for consistency among programs.

Reply:
I just figured out that the 'atommap' goes reverse mapping from target to original in direction from the mill.  I've got the atom sorting working now.